src/Donations/Controllers/DonationRequestController.php

```diff
@@ -0,0 +1,210 @@
+<?php
+
+namespace Give\Donations\Controllers;
+
+use Give\Donations\Models\Donation;
+use Give\Donations\ValueObjects\DonationAnonymousMode;
+use Give\Donations\ValueObjects\DonationRoute;
+use WP_REST_Request;
+use WP_REST_Response;
+
+/**
+ * @unreleased
+ */
+class DonationRequestController
+{
+    /**
+     * @unreleased
+     */
+    public function getDonation(WP_REST_Request $request): WP_REST_Response
+    {
+        $donation = Donation::find($request->get_param('id'));
+
+        if ( ! $donation) {
+            return new WP_REST_Response(
+                ['message' => __('Donation not found', 'give')],
+                404
+            );
+        }
+
+        $isAdmin = current_user_can('manage_options');
+
+        $includeSensitiveData = $request->get_param('includeSensitiveData');
+        if ( ! $isAdmin && $includeSensitiveData) {
+            return new WP_REST_Response(
+                ['message' => __('You do not have permission to include sensitive data.', 'give')],
+                403
+            );
+        }
+
+        $donationAnonymousMode = new DonationAnonymousMode($request->get_param('anonymousDonations'));
+        if ( ! $isAdmin && $donation->anonymous && ! $donationAnonymousMode->isRedacted()) {
+            return new WP_REST_Response(
+                ['message' => __('You do not have permission to include anonymous donations.', 'give')],
+                403
+            );
+        }
```

Comment:
@glaubersilva This shouldn't be happening here, to tie this together. The request handler should be able to assume that all authentication and such has happened. It's concern is purely in handling and returning the request appropriately.

Reply:
Resolved in 0bc52d67abe9047987a915fc436ddab1cfbc91c4